modules/migration/migration-5.2.0_to_5.3.0/wso2-is-migration-client/src/main/java/org/wso2/carbon/is/migration/dao/IdpMetaDataDAO.java

```diff
@@ -0,0 +1,190 @@
+/*
+ * Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wso2.carbon.is.migration.dao;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.core.util.IdentityDatabaseUtil;
+import org.wso2.carbon.is.migration.ISMigrationException;
+import org.wso2.carbon.is.migration.SQLConstants;
+import org.wso2.carbon.utils.DBUtils;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Data Access layer for IDP Metadata table
+ */
+public class IdpMetaDataDAO {
+
+    private static Log log = LogFactory.getLog(IdpMetaDataDAO.class);
+
+    private static IdpMetaDataDAO idpMetaDataDAO = new IdpMetaDataDAO();
+
+    private IdpMetaDataDAO() {
+
+    }
+
+    public static IdpMetaDataDAO getInstance() {
+
+        return idpMetaDataDAO;
+    }
+
+
+    public int getResidentIdpId(int tenantId) throws ISMigrationException {
+
+        // we use the "RememberMeTimeout" attribute as a way to identify the resident IDP id.
+        String sql = "SELECT IDP_ID FROM IDP_METADATA WHERE TENANT_ID=? AND NAME=\"RememberMeTimeout\";";
```

Comment:
AFAIK, IDP.Name is 'LOCAL' for resident IdPs. We can use that instead of this property.